cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go

```diff
@@ -146,6 +148,10 @@ func (etp *EthTxPool) Nonce(addr common.Address) uint64 {
 			pendingNonces[addr]++
 		case txNonce > pendingNonce+1:
 			// As soon as we see a non contiguous nonce we break.
+			exit = true
+		}
+
+		if exit {
```

Comment:
you can actually use a "label" to do this 
<img width="504" alt="image" src="https://github.com/berachain/polaris/assets/113940816/df4fb2c7-2012-4973-bf72-373bcd44bb04">